diff --git a/src/graphql/client.ts b/src/graphql/client.ts
index 65d953a9edca56789ec8351b85ba804aef0c1978..7b3f8a623891a886e970d6377ccf7c0c46ba4f98 100644
--- a/src/graphql/client.ts
+++ b/src/graphql/client.ts
@@ -19,6 +19,7 @@ export type GraphQLQueryOptions<
 	Result = Record<string, unknown>,
 	Variables = Record<string, unknown>,
 > = {
+	// @ts-ignore
 	query: GraphQLDocument<Result, Variables>;
 	operationName?: string;
 	extensions?: Record<string, unknown>;
@@ -78,6 +79,7 @@ export class SuiGraphQLClient<Queries extends Record<string, GraphQLDocument> =
 				...this.#headers,
 			},
 			body: JSON.stringify({
+				// @ts-ignore
 				query: typeof options.query === 'string' ? String(options.query) : print(options.query),
 				variables: options.variables,
 				extensions: options.extensions,
diff --git a/src/zklogin/publickey.ts b/src/zklogin/publickey.ts
index a646c00d57aabb16556cf55dd804302cd33ad512..50d83318c486f5d44dfa788f8de96cecbb43b9ea 100644
--- a/src/zklogin/publickey.ts
+++ b/src/zklogin/publickey.ts
@@ -152,6 +152,7 @@ async function graphqlVerifyZkLoginSignature({
 	client?: SuiGraphQLClient;
 }) {
 	const resp = await client.query({
+		// @ts-ignore
 		query: VerifyZkLoginSignatureQuery,
 		variables: {
 			bytes,
@@ -162,7 +163,9 @@ async function graphqlVerifyZkLoginSignature({
 	});
 
 	return (
+		// @ts-ignore
 		resp.data?.verifyZkloginSignature.success === true &&
+		// @ts-ignore
 		resp.data?.verifyZkloginSignature.errors.length === 0
 	);
 }
